components/movementsensor/replay/replay_test.go

```diff
@@ -262,7 +261,13 @@ func TestNewReplayMovementSensor(t *testing.T) {
 			replay, _, serverClose, err := createNewReplayMovementSensor(ctx, t, tt.cfg, tt.validCloudConnection)
 			if tt.expectedErr != nil {
 				test.That(t, err, test.ShouldNotBeNil)
-				test.That(t, err, test.ShouldBeError, tt.expectedErr)
+				if strings.Contains(err.Error(), "rpc error: code = DeadlineExceeded") {
```

Comment:
This might mask an issue where we're not getting the error we're expecting. For example, if we're expecting the error "invalid time format for start time, missed during config validation", and we get a DeadlineExceeded error instead, then we're failing to test for the invalid time format case. Is it possible to make this the `expectedErr` for the cases where it occurs? Or at least to only allow the DeadlineExceeded error in particular cases, such as when the `expectedErr` is `errPropertiesFailedToInitializeTest`?

Reply:
That's a good point. We can predict when this error occurs as opposed to `errPropertiesFailedToInitializeTest`, because getting we're getting the `DeadlineExceeded` error when [this context times out](https://github.com/viamrobotics/rdk/blob/0f4c3b08d7bb16d3c8562cd1f564a1ec90fe558a/components/movementsensor/replay/replay.go#L474) while we're in the call to [`TabularDataByFilter`](https://github.com/viamrobotics/rdk/blob/0f4c3b08d7bb16d3c8562cd1f564a1ec90fe558a/components/movementsensor/replay/replay.go#L474).

After discussing this with @jeremyrhyde, though, we realized that his already merged fix for [this bug](https://viam.atlassian.net/browse/RSDK-5457) solves the problem, since we're now using a separate timeout for `TabularDataByFilter`.

We found a few bugs, though, which I'll fix in this PR.